### PR TITLE
add `closeOnEscape` option to calcite-modal

### PR DESCRIPTION
**Pull request by vcolavin**
The code here is hopefully self-explanatory.

I have a use-case where the user can do a bunch of work inside of the modal, and accidentally hitting "escape" would dismiss all of that work without a prompt, which I've already found is very frustrating.

Even with this option, the modal ought to remain accessible because the user can still tab to the "X" button and hit enter.

### CODE REVIEW

**Comment by paulcpederson** [comment]
@vcolavin could be good to add a test for this prop

**Comment by vcolavin** [comment]
@paulcpederson 
Good idea. I'm writing the test now. It's a good thing I'm doing this because I've found a problem.

Because of the way HTML boolean attributes work, I think it might be much simpler to have a `disableCloseOnEscape` which defaults to `false`, rather than a `closeOnEscape` which defaults to `true`. I don't like the ergonomics of this as much but I can't think of a simpler solution.

Basically, `<calcite-modal closeOnEscape="false"></calcite-modal>` still has a truthy value, because you can't pass in `false` itself, just a string. The only way for it to be `false` is if the attribute is not present, but even then it still defaults to `true`.

@macandcheese thoughts?

**Comment by macandcheese** [comment]
That makes sense, in html (vs. jsx) it'd be `disable-close-on-escape="true"` as attribute to **not** allow users to escape to close, yeah?

**Comment by vcolavin** [comment]
Right... phrasing the attribute as a negative ("set to true if you *don't* want this") is kind of a code smell to me.

**Comment by vcolavin** [comment]
@macandcheese if the property/attribute inside the component is `disableCloseOnEscape`, does it become `disable-close-on-escape` when you're writing plain HTML?

**Comment by macandcheese** [comment]
Yeah, it becomes dashed - https://stenciljs.com/docs/properties. You could change it using the "attribute" option but, doesn't really solve for anything above 🤷‍♂

**Comment by paulcpederson** [comment]
@vcolavin I would suggest not using `="true"`. Just the presence of the attribute should denote `true`. This is very similar to `disabled` on a form element. That is a `negative=true` but using just the attribute makes it not weird

**Comment by vcolavin** [comment]
@paulcpederson thanks, that's super helpful.

**Comment by paulcpederson** [comment]
probably a better name for this attribute would be `disable-escape`

**Comment by paulcpederson** [comment]
💯  LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### [Unreleased][HEAD]
 
+## [v1.0.0-beta.8] - Sep 3rd 2019
+
+### Added
+
+* Adds a boolean "closeOnEscape" prop to calcite-modal to make closing on escape optional.
```
Comment (paulcpederson) [inline CHANGELOG.md]: this is out of date now, I think

**Comment by vcolavin** [comment]
Just for posterity, here's a bug I found while exploring this issue:

https://github.com/Esri/calcite-components/issues/130
